### PR TITLE
Fixes #3341 Add Nutri-Score in products to be completed

### LINKED ISSUE #3341: Add the Nutri-Score in the "Products to be completed" activity by refactoring it
- [ ] Add the Nutri-Score in the "products to be completed" activity so that power users can see which new products have a Nutri-Score, and which ones don't have one
- [ ] Get the occasion to standardize the activity with the other list activities that already show the Nutri-Score

### PR DESCRIPTION
**Pull request by Varsha-29**
The nutri score is populated in the additional properties with "nutrition_grade_tags" as key. But to display the score, only NutritionGradeFr was considered. Modified the method to also check the additional properties of the product to check for the score

### CODE REVIEW

**Comment by teolemon** [comment]
Both those fields are the exact same value: https://world.openfoodfacts.org/api/v0/product/5410041001204.json?fields=nutrition_grade_fr,nutrition_grades_tags
It's just some API mess on the server side. I guess the issue is purely a display one.

**Comment by Varsha-29** [comment]
These are the fields which are used in querying the response for incomplete products API : "brands,product_name,image_small_url,quantity,nutrition_grades_tags,code"
Because of this "nutrition_grade_fr" is null in the Product model.

**Comment by Varsha-29** [comment]
@teolemon , could you let me know if anything has to be changed as a part of this PR?

**Comment by teolemon** [comment]
It seems to behave correctly. @VaiTon ok to merge codewise ?

**Comment by VaiTon** [comment]
Checking and then merging.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java [hunk 1]
```diff
@@ -355,7 +355,11 @@ public static int getNovaGroupDrawable(@Nullable String novaGroup) {
     }
 
     public static int getSmallImageGrade(Product product) {
-        return getSmallImageGrade(product == null ? null : product.getNutritionGradeFr());
+        String nutritionGrade = null;
+        if (product != null) {
+            nutritionGrade = product.getNutritionGradeFr() != null ? product.getNutritionGradeFr() : product.getNutritionGradeTag();
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java:360]: Why do we prefer the FR grade? Any particular reason? @teolemon @Varsha-29

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java [hunk 2]
```diff
@@ -355,7 +355,11 @@ public static int getNovaGroupDrawable(@Nullable String novaGroup) {
...
+        }
+        return getSmallImageGrade(nutritionGrade);
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java:362]: Can we inline this return? I prefer having multiple exit points than having to remember the value of `nutritionGrade`.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java (hunk 1 above)
Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java:360]: That's the name we initially gave to the Nutri-Score when it didn't have a name.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java (hunk 2 above)
Comment (Varsha-29) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java]: Done

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java (hunk 1 above)
Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java]: Ok, but here we prefer using the FR tag. Are the FR tag and the "global" the same? Any differences?

**Pull request merged**